python_modules/libraries/dagster-dlt/dagster_dlt/components/dlt_load_collection/component.py

```diff
@@ -104,7 +73,14 @@ class DltLoadSpecModel(Resolvable):
             model_field_type=str,
         ),
     ]
-    translation: Optional[ResolvedTranslationFn] = None
+    translation: Optional[
+        Annotated[
+            TranslationFn[DltResourceTranslatorData],
+            TranslationFnResolver[DltResourceTranslatorData](
```

Comment:
the double `[DltResourceTranslatorData]` is little unfortunate - the latter one does catch mistakes in the data function though right?